src/components/dock/dock.scss

```diff
@@ -70,7 +70,7 @@ nav {
     flex-grow: 1;
 
     padding: var(--dock-padding); // needed for the focus effect
-
+    padding-bottom: calc((var(--dock-padding) * 2.4));
     overflow-y: auto;
```

Comment:
The shadow cut-off appears due to the `overflow` on the scrollable area of the Dock. The overflow is needed, because there could be many items in this list. However, a better fix might be doing something about that. Now I don't want to give you the solution, because this is still about practicing and understanding how everything is connected, and what concerns exist in each component. So, I'm curious to see what ideas you can come up with.

Reply:
For a little clarification: the dock should remain 100vh and overflow for more possible items but not cut anything off when it's just the two default in the footer? Or will the footer never grow higher?

Reply:
I'm not sure if I understood your question. Could you please describe more?